### DISCUSSION TITLE
How many meshes geosx can support at its maximum?

### DISCUSSION THREAD

**Discussion by FishYNY** [GEOS-DEV/GEOS#2302] [Usability]
How many meshes geosx can support at its maximum? We faced with the problem of  computing with the hundreds of million grids. Faced with such a large number of grids, we will consider use a multi-million core scale platform for computing. So I wonder if geosx can still support it.

**Comment by joshua-white** [comment]
@FishYNY, do you mean (1) a single mesh with hundreds of millions of grid cells, or (2) a simulation involving hundreds of millions of separate grids?  We can certainly support (1).  For (2), it seems like you could use a wrapper script to execute GEOS many times over the individual grids, with each simulation a completely separate instance.  Does your use case correspond to (1) or (2), or possibly something different?

**Reply by FishYNY** [reply]
The case correspond to (2). The mesh is exported from the field cases, maybe including reservoir, overburden, underburden, fault and fracture grids. So the number of grids is completely huge.

**Reply by joshua-white** [reply]
In that case, I would suggest writing a wrapper script that executes a separate instance of GEOS each time, looping over a series of input meshes.  since these runs are independent of one another, they could be handled in parallel on your cluster.  the script could then post process the output to produce whatever integrated results you need.
